include/eve/algo/container/detail/soa_storage2.hpp

```diff
@@ -0,0 +1,199 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/algo/views/zip.hpp>
+#include <eve/arch/expected_cardinal.hpp>
+#include <eve/concept/simd_allocator.hpp>
+#include <eve/detail/kumi.hpp>
+#include <eve/memory/aligned_allocator.hpp>
+#include <eve/module/core.hpp>
+#include <eve/product_type.hpp>
+#include <cstddef>
+#include <type_traits>
+#include <memory>
+
+namespace eve::algo::detail
+{
+  struct memory_helper
+  {
+    static void copy_all_aligned(std::byte const* src, std::byte* dst, std::size_t size)
+    {
+      auto aligned_src = eve::as_aligned(reinterpret_cast<std::uint8_t const*>(src), lane<64>);
+      auto aligned_dst = eve::as_aligned(reinterpret_cast<std::uint8_t*>(dst), lane<64>);
+      eve::algo::copy(as_range(aligned_src, aligned_src+size), aligned_dst);
+    }
+
+    template<typename T> static EVE_FORCEINLINE auto as_aligned_pointer(T* ptr)
+    {
+      return eve::as_aligned(ptr, eve::detail::cache_line_cardinal<T>{});
+    }
+
+    template<typename T>
+    static  auto realign(std::size_t n) { return eve::align(sizeof(T)*n, eve::over{64}); }
+
+    template<typename Storage>
+    static auto allocate(Storage const& s, std::size_t n)
+    {
+      using storage_t = typename Storage::storage_t;
```

Comment:
storage::storage_t ? That's at least weird.

Reply:
indeed, gonna rename

Reply:
renamed to data_t and data_